File: src/co/edu/uptc/music/servlets/LoginServlet.java
package co.edu.uptc.music.servlets;

import com.google.gson.Gson;

import java.io.IOException;
import java.io.PrintWriter;
import java.security.MessageDigest;

import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import javax.xml.bind.DatatypeConverter;

import co.edu.uptc.music.logic.managers.UsersManager;
import co.edu.uptc.music.logic.models.User;
import co.edu.uptc.music.logic.models.UserType;

@WebServlet(name = "LoginServlet", urlPatterns = {"/LoginServlet"})
public class LoginServlet extends HttpServlet {

    private UsersManager usersManager = new UsersManager();

    public void processRequest(HttpServletRequest request, HttpServletResponse response) {
        response.setContentType("application/json");
        response.setCharacterEncoding("UTF-8");
        try {
            usersManager.load();

            Gson gson = new Gson();
            try (PrintWriter writer = response.getWriter()) {
                String name = request.getParameter("username");
                String pass = request.getParameter("password");
                String login = request.getParameter("login");

                int loginValue = Integer.parseInt(login);

                if (loginValue == 1) {
                    User user = usersManager.findItem(name);

                    if (user != null) {
                        MessageDigest digest = MessageDigest.getInstance("MD5");

                        digest.update(pass.getBytes());
                        byte[] chainAux = digest.digest();
                        String myHash = DatatypeConverter.printHexBinary(chainAux);

                        if (user.validateUser(myHash)) {
                            String aux = gson.toJson(user);
                            StringBuilder sb = new StringBuilder();
                            sb.append("{\"code\":2,");
                            sb.append(aux.substring(1, aux.length() - 1));
                            if (user.getType() == UserType.ADMIN) {
                                String usersList = gson.toJson(usersManager.getList());
                                sb.append(",\"list\": ").append(usersList);
                            }
                            sb.append("}");
                            writer.print(sb.toString());
                        } else {
                            writer.print("{\"code\":1,\"error\":\"La contraseña es incorrecta.\"}");
                        }
                    } else {
                        writer.print("{\"code\":0,\"error\":\"El usuario no se encuentra " +
                                "registrado" +
                                ".\"}");
                    }
                } else if (loginValue == 2) {
                    String fullname = request.getParameter("fullname");
                    String email = request.getParameter("email");
                    if (usersManager.addNewUser(fullname, email, name, pass)) {
                        writer.print("{\"code\": 4}");
                    } else {
                        writer.print("{\"code\": 3, \"error\": \"El usuario ya se encuentra " +
                                "registrado en la base de datos\"}");
                    }
                }
                writer.close();
            }
        } catch (Exception ignored) {
        }
    }

    @Override
    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws
            ServletException, IOException {
        processRequest(req, resp);
    }

    @Override
    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws
            ServletException, IOException {
        processRequest(req, resp);
    }

    @Override
    public String getServletInfo() {
        return super.getServletInfo();
    }

}

File: src/co/soundwave/servlets/LoginServlet.kt
package co.soundwave.servlets

import co.soundwave.extensions.ignore
import co.soundwave.managers.ArtistsManager
import co.soundwave.managers.UsersManager
import com.google.gson.Gson
import java.security.MessageDigest
import javax.servlet.annotation.WebServlet
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse
import javax.xml.bind.DatatypeConverter

@WebServlet(name = "LoginServlet", urlPatterns = ["/LoginServlet"])
class LoginServlet : BaseServlet() {
    
    private val usersManager = UsersManager()
    private val artistsManager = ArtistsManager()
    
    override fun processRequest(
        request: HttpServletRequest,
        response: HttpServletResponse,
        type: Type
                               ) {
        response.contentType = "application/json"
        response.characterEncoding = "UTF-8"
        ignore {
            usersManager.load()
            artistsManager.load()
            
            val gson = Gson()
            response.writer.use { writer ->
                val nick = request.getParameter("username")
                val pass = request.getParameter("password")
                val login = request.getParameter("login")
                
                val loginValue = Integer.parseInt(login)
                
                if (loginValue == 1) {
                    val user = usersManager.findItem(nick)
                    
                    val digest = MessageDigest.getInstance("MD5")
                    digest.update(pass.toByteArray())
                    val chainAux = digest.digest()
                    val myHash = DatatypeConverter.printHexBinary(chainAux)
                    val inputPassword = try {
                        myHash.substring(0, 40)
                    } catch (e: Exception) {
                        myHash
                    }
                    
                    if (user != null) {
                        if (user.validate(inputPassword)) {
                            val aux = gson.toJson(user)
                            val sb = StringBuilder()
                            sb.append("{\"code\":2,")
                            sb.append(aux, 1, aux.length - 1)
                            val usersList = gson.toJson(usersManager.getList())
                            sb.append(",\"list\": ").append(usersList)
                            sb.append("}")
                            writer.print(sb.toString())
                        } else {
                            writer.print("{\"code\":1,\"error\":\"La contraseña es incorrecta.\"}")
                        }
                    } else {
                        val artist = artistsManager.findItem(nick)
                        
                        if (artist != null) {
                            if (artist.validate(inputPassword)) {
                                val aux = gson.toJson(artist)
                                val sb = StringBuilder()
                                sb.append("{\"code\":4,")
                                sb.append(aux, 1, aux.length - 1)
                                val usersList = gson.toJson(usersManager.getList())
                                sb.append(",\"list\": ").append(usersList)
                                sb.append("}")
                                writer.print(sb.toString())
                            } else {
                                writer.print(
                                    "{\"code\":1,\"error\":\"La contraseña es incorrecta.\"}")
                            }
                        } else {
                            writer.print(
                                "{\"code\":0,\"error\":\"El usuario no se encuentra " + "registrado.\"}")
                        }
                    }
                } else if (loginValue == 2) {
                    val doc = Integer.parseInt(request.getParameter("doc"))
                    val name = request.getParameter("name")
                    val lastName = request.getParameter("lastName")
                    val age = Integer.parseInt(request.getParameter("age"))
                    val email = request.getParameter("email")
                    // if (usersManager.addNewUser(fullname, email, name, pass)) {
                    if (usersManager.addNewUser(doc, name, lastName, age, nick, email, pass)) {
                        writer.print("{\"code\": 4}")
                    } else {
                        writer.print(
                            "{\"code\": 3, \"error\": \"El usuario ya se encuentra " + "registrado en la base de datos\"}")
                    }
                }
                writer.close()
            }
        }
    }
    
    override fun getServletInfo(): String = "Login servlet"
}